### ISSUE TITLE
RF: Track all branches

### ISSUE THREAD

**Issue by alaegin** [janbarari/gradle-analytics-plugin#52]
Hello!

Is it possible to track everything by default? We have a lot of branches and a lot of CI tasks on them. 
We would like to track everything and then get detailed statistics about specific tasks executed on **all** branches. 

Is it a good idea? If not, could you suggest another way of analyzing tasks on different branches?

**Comment by janbarari** [comment]
Hey @alaegin 
Thank you for sending the feedback.

Based on your question, I need to share some information about Gradle tasks.

Actually, everything in Gradle is done by a task. So, in that case, it is not a good idea to have an analysis for every task of a project. because It creates a lot of junk results in the database and affects your daily build process speed.

On the other hand, many Git branches are temporary, and providing analysis for them won't get accurate results.

I would suggest you add your `requested tasks` and `branches` in the plugin configuration.
You can add as many tasks and branches as you want.

**Comment by alaegin** [comment]
Thank you for your answer, @janbarari!

The explanation about tasks sounds reasonable, I didn't take this into account.

What about branches: I agree that we could have a lot of short-living branches but in our case, the development process takes part in feature branches. We would like to analyze them.

There could be 10-20-50 such branches at the same time and it seems impossible to maintain this huge list. 
Of course, we can maintain this list dynamically and use an environment variable for adding the current branch on CI.
Something like this:
```kotlin
trackingBranches = listOf(System.getenv("CI_COMMIT_REF_NAME"))
```

What about giving the ability to analyze all branches with an additional property?

**Comment by janbarari** [comment]
Since I have to realize the side effects of this feature, I will put this conversation into an investigation and will get back to you here with more details.

If it was safe feature. I will move the conversation to `Request Feature`.

**Comment by alaegin** [comment]
Thanks!

If it's not safe, it would be nice to have an ability to request branch analytics by regexp.

**Comment by janbarari** [comment]
I will consider this option too. Thanks for mentioning :)

**Comment by janbarari** [comment]
Tracking all branches is in the development process. It will be released in `v1.0.0-beta6`.
